consensus/hotstuff/vote_collector.go

```diff
@@ -0,0 +1,28 @@
+package hotstuff
+
+import (
+	"github.com/onflow/flow-go/consensus/hotstuff/model"
+	"github.com/onflow/flow-go/model/flow"
+)
+
+// VoteCollector collects votes for the same block, produces QC when enough votes are collected
+// VoteCollector takes a callback function to report the event that a QC has been produced.
+type VoteCollector interface {
+	// AddVote adds a vote to the collector
+	// returns true if the vote was added
+	// returns false otherwise
+	// return error if the signature is invalid
+	// When enough votes have been added to produce a QC, the QC will be created asynchronously, and
+	// passed to EventLoop through a callback.
+	AddVote(vote *model.Vote) (bool, error)
```

Comment:
Consider `VerifyAndAddVote`?

Reply:
see the reasoning here https://github.com/onflow/flow-go/pull/1035#discussion_r680282500